Mathlib/Lemma.lean

```diff
@@ -0,0 +1,3 @@
+-- this macro, written by Sebastian Ullrich, enables us to have lemmas
+macro mods:declModifiers "lemma" n:declId sig:declSig val:declVal : command => `($mods:declModifiers theorem $n $sig $val)
```

Comment:
Two things:
1. we already have this
2. you need to add every file to `Mathlib.lean` now because leanpkg only builds its dependencies nowadays.  @kha Is that intentional?
3. I can't import both `Mathlib.Lemma` and `Mathlib.Tactic.Basic` now because of the error `import failed, environment already contains 'command_Lemma___._closed_6._cstage2'`.  Since you're already here @kha, is this is a bug?

Reply:
> you need to add every file to Mathlib.lean now because leanpkg only builds its dependencies nowadays. @Kha Is that intentional?

Yes, like in Rust (if I'm not mistaken). It does have the nice effect that stray files (e.g. from a change otherwise stashed away) don't break your build.

Reply:
> I can't import both Mathlib.Lemma and Mathlib.Tactic.Basic now

Sadly it seems unavoidable if we want to give parsers reasonable names (so we can `open` them)

Reply:
Rust will follow transitive dependencies though. That would probably work better for us since we already have the `default.lean` files which import everything in a subdirectory.

Reply:
I think the name collision is acceptable. The workaround is to manually name the macro/syntax, or just to not have two copies of the macro (which will cause other problems anyway).

Reply:
> Yes, like in Rust (if I'm not mistaken). It does have the nice effect that stray files (e.g. from a change otherwise stashed away) don't break your build.

Maybe we need to add a script to update `Mathlib.lean` then.  One way or another, CI must complain if somebody adds a file that doesn't build.

> Rust will follow transitive dependencies though

The new leanpkg does as well, but this was not a transitive dependency of anything else.. because it's a brand new file.

Reply:
> Sadly it seems unavoidable if we want to give parsers reasonable names (so we can open them)

The only surprising thing is that it works just fine if it's in the same file.  But this is just something to keep in mind when refactoring.

```lean
macro mods:declModifiers "lemma" n:declId sig:declSig val:declVal : command =>
  `($mods:declModifiers theorem $n $sig $val)
macro mods:declModifiers "lemma" n:declId sig:declSig val:declVal : command =>
  `($mods:declModifiers theorem $n $sig $val)
```